src/subprocess-posix.cc

```diff
@@ -330,9 +351,81 @@ void SubprocessSet::Clear() {
     // Since the foreground process is in our process group, it will receive
     // the interruption signal (i.e. SIGINT or SIGTERM) at the same time as us.
     if (!(*i)->use_console_)
-      kill(-(*i)->pid_, interrupted_);
+      Xkill(-(*i)->pid_, interrupted_);
   for (vector<Subprocess*>::iterator i = running_.begin();
        i != running_.end(); ++i)
     delete *i;
   running_.clear();
 }
+
+void SubprocessSet::Suspend() {
+  assert(IsSuspended());
+
+  // errno may change during the course of this function.
+  int saved_errno = errno;
+
+  // Suspend children.
+  for (vector<Subprocess*>::iterator i = running_.begin();
+       i != running_.end(); ++i)
+    if (!(*i)->use_console_)
+      Xkill(-(*i)->pid_, SIGTSTP);
+
+  // Tell users what we have done.
+  {
+    // We use write to bypass the buffer that may not be flush before
+    // we re-raise the SIGTSTP for us.
+    const char* msg = "\nninja: entire build suspended.\n";
+    if (write(STDOUT_FILENO, msg, strlen(msg)) < 0) {
+      // If the write fails, there's nothing we can do.
+      // But this block seems necessary to silence the warning.
```

Comment:
If I recall correctly, the canonical way of silencing the warning is to use:

    (void) write(...);